### PR TITLE
check pod default before post request

### PR DESCRIPTION
**Pull request by lluunn**
Verified that 2nd apply succeeded with the fix.

fix https://github.com/kubeflow/kubeflow/issues/4397

/cc @kunmingg @jlewi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4398)

### CODE REVIEW

File: bootstrap/pkg/kfapp/gcp/gcp.go [hunk 1]
```diff
@@ -1881,6 +1881,12 @@ func (gcp *Gcp) ConfigPodDefault() error {
 		}
 	}
 
+	getReq := crdClient.Get().Resource(mapping.Resource.Resource).Namespace(defaultNamespace).Name("add-gcp-secret")
```
Comment (jlewi) [inline bootstrap/pkg/kfapp/gcp/gcp.go]: Nit define a constant for the name and use it here and here
https://github.com/kubeflow/kubeflow/blob/0a370635ad1db8139a1cf56bef329f34344fea6d/bootstrap/pkg/kfapp/gcp/gcp.go#L1761

Comment (lluunn) [reply bootstrap/pkg/kfapp/gcp/gcp.go]: done

**Comment by jlewi** [comment]
@abhi-g This is for backwards compatibility but it is in the default namespace created by kfctl. This matches 0.6 behavior so if someone is using PodDefault in that namespace things would still work.

/lgtm
/approve
/hold cancel

**Pull request merged**